src/BlockTypes/FeaturedItem.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+/**
+ * FeaturedItem class.
+ */
+abstract class FeaturedItem extends AbstractDynamicBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name;
+
+	/**
+	 * Default attribute values.
+	 *
+	 * @var array
+	 */
+	protected $defaults = array(
+		'align' => 'none',
+	);
+
+	/**
+	 * Global style enabled for this block.
+	 *
+	 * @var array
+	 */
+	protected $global_style_wrapper = array(
+		'text_color',
+		'font_size',
+		'border_color',
+		'border_radius',
+		'border_width',
+		'background_color',
+		'text_color',
+		'padding',
+	);
+
+	/**
+	 * Returns the featured item.
+	 *
+	 * @param array $attributes Block attributes. Default empty array.
+	 * @return \WP_Term|\WC_Product|null
+	 */
+	abstract protected function get_item( $attributes );
+
+	/**
+	 * Returns the name of the featured item.
+	 *
+	 * @param \WP_Term|\WC_Product $item Item object.
+	 * @return string
+	 */
+	abstract protected function get_item_title( $item);
+
+	/**
+	 * Returns the featured item image URL.
+	 *
+	 * @param \WP_Term|\WC_Product $item Item object.
+	 * @param string               $size    Image size, defaults to 'full'.
```

Comment:
I'm not sure about this, but are PHPDocs supposed to be aligned, and how? The description of `size` is off compared to the one above.

Reply:
Yeah, I'm having some issue with autoformatting, I need to look into it 🤔